x-pack/plugins/apm/server/routes/mobile/get_mobile_filters.ts

```diff
@@ -50,79 +35,49 @@ export async function getMobileFilters({
   start: number;
   end: number;
 }): Promise<MobileFiltersResponse> {
-  const response = await apmEventClient.search('get_mobile_filters', {
-    apm: {
-      events: [ProcessorEvent.transaction],
-    },
-    body: {
-      track_total_hits: false,
-      size: 0,
-      query: {
-        bool: {
-          filter: [
-            ...termQuery(SERVICE_NAME, serviceName),
-            ...termQuery(TRANSACTION_TYPE, transactionType),
-            ...rangeQuery(start, end),
-            ...environmentQuery(environment),
-            ...kqlQuery(kuery),
-          ],
-        },
-      },
-      aggs: {
-        devices: {
-          terms: {
-            field: DEVICE_MODEL_IDENTIFIER,
-            size: 10,
-          },
-        },
-        osVersions: {
-          terms: {
-            field: HOST_OS_VERSION,
-            size: 10,
-          },
-        },
-        appVersions: {
-          terms: {
-            field: SERVICE_VERSION,
-            size: 10,
-          },
-        },
-        netConnectionTypes: {
-          terms: {
-            field: NETWORK_CONNECTION_TYPE,
-            size: 10,
-          },
-        },
-      },
-    },
-  });
+  const MAX_ITEMS = 10;
+  const commonProps = {
+    kuery,
+    apmEventClient,
+    serviceName,
+    transactionType,
+    environment,
+    start,
+    end,
+    size: MAX_ITEMS,
+  };
+  const mobileTransactionEventsFiltersResponse = await getDeviceOSApp(
+    commonProps
+  );
+
+  const mobileNetworkConnectionTypeFiltersResponse = await getNCT(commonProps);
```

Comment:
We can use a Promise.all here

Reply:
done here https://github.com/elastic/kibana/pull/162794/commits/358cddff54bf5d82d2e869acecdf29224c43345f